tests/test_lane.py

```diff
@@ -2249,67 +2473,52 @@ def test_customlist_filter_clauses(self):
         # Gutenberg.
-        works_on_gutenberg_lists = WorkList()
+        works_on_gutenberg_lists = DatabaseBackedWorkList()
         works_on_gutenberg_lists.initialize(
             self._default_library, list_datasource=gutenberg
         )
-        self.add_to_materialized_view([work])
 
         def _run(qu, clauses):
-            # Run a query with certain clauses and pick out the
-            # work IDs returned.
-            modified = qu.filter(and_(*clauses))
-            return [x.works_id for x in modified]
+            # Run a query with certain clauses
+            return qu.filter(and_(*clauses)).all()
 
         def results(wl=works_on_gutenberg_lists, must_be_featured=False):
-            qu = self._db.query(work_model)
-            new_qu, clauses = wl.customlist_filter_clauses(
-                qu, must_be_featured=must_be_featured
-            )
+            qu = wl.base_query(self._db)
+            new_qu, clauses = wl.customlist_filter_clauses(qu)
 
-            if must_be_featured or wl.list_seen_in_previous_days:
-                # The query comes out different than it goes in -- there's a
-                # new join against CustomListEntry.
-                assert new_qu != qu
+            # The query comes out different than it goes in -- there's a
+            # new join against CustomListEntry.
+            assert new_qu != qu
             return _run(new_qu, clauses)
 
         # Both lanes contain the work.
-        eq_([work.id], results(works_on_list))
-        eq_([work.id], results(works_on_gutenberg_lists))
+        eq_([work], results(works_on_list))
+        eq_([work], results(works_on_gutenberg_lists))
 
         # If there's another list with the same work on it, the
         # work only shows up once.
         gutenberg_list_2, ignore = self._customlist(num_entries=0)
         gutenberg_list_2_entry, ignore = gutenberg_list_2.add_entry(work)
         works_on_list._customlist_ids.append(gutenberg_list.id)
-        eq_([work.id], results(works_on_list))
+        eq_([work], results(works_on_list))
 
         # This WorkList gets every work on a list associated with Overdrive.
         # There are no such lists, so the lane is empty.
         overdrive = DataSource.lookup(self._db, DataSource.OVERDRIVE)
-        works_on_overdrive_lists = WorkList()
+        works_on_overdrive_lists = DatabaseBackedWorkList()
         works_on_overdrive_lists.initialize(
             self._default_library, list_datasource=overdrive
         )
         eq_([], results(works_on_overdrive_lists))
 
-        # It's possible to restrict a WorkList so that only works that
-        # are _featured_ on a list show up. The work isn't featured,
-        # so it doesn't show up.
-        eq_([], results(must_be_featured=True))
-
-        # Now it's featured, and it shows up.
-        gutenberg_list_entry.featured = True
-        eq_([work.id], results(must_be_featured=True))
-
```

Comment:
Again, this feature was removed because it was only used in the context of generating a grouped feed for a Lane -- something that's now handled through the search index.